src/controller/CustomersController.ts

```diff
@@ -18,6 +19,18 @@ class CustomersController {
       next(ApiError.internal(message));
     }
   }
+
+  async getPetGift(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { customerId } = req.params; 
+      const promotionId = "6MonthPurchase";
```

Comment:
I stored the promotion code/ id here since the express route didn't specify this information. However, I removed it and integrated it into the route on the suggestion branch.